llama_index/indices/base.py

```diff
@@ -340,30 +342,65 @@ def as_query_engine(self, **kwargs: Any) -> BaseQueryEngine:
         return RetrieverQueryEngine.from_args(**kwargs)
 
     def as_chat_engine(
-        self, chat_mode: ChatMode = ChatMode.CONDENSE_QUESTION, **kwargs: Any
+        self, chat_mode: ChatMode = ChatMode.BEST, **kwargs: Any
     ) -> BaseChatEngine:
+        query_engine = self.as_query_engine(**kwargs)
+        if "service_context" not in kwargs:
```

Comment:
this is a good fix, people often before would pass it in and get errors about it being included twice